### ISSUE TITLE
Inline convert to `Prefixed`

### ISSUE THREAD

**Issue by dan-fritchman** [dan-fritchman/Hdl21#61]
Originally came up as part of #45, along with the realization that most `Primitive` parameters should be of type `Prefixed`.  
It would be nice to be able to "inline" convert to `Prefixed` from built-in primitive types, especially `int`. So that stuff like: 

```python
@h.paramclass 
class Params:
  num = h.Param(dtype=h.Prefixed, desc="watch")

Params(num = Prefixed(1000, MILLI)) # Works
Params(num = 11 * µ) # Works

Params(num = 1) # <= this here
```

Would work. 

I filed a Pydantic issue at the time: 
https://github.com/pydantic/pydantic/issues/4625

And the author's recommendation was to use their custom data types feature: 
https://pydantic-docs.helpmanual.io/usage/types/#custom-data-types. 

It didn't immediately jump off the page how to fit those together.  
This issue: make it work.

**Comment by dan-fritchman** [comment]
OK I get what the Pydantic authors were recommending now.  
And it doesn't look like this will take much effort.  
Stuff like this, works: 

```python
from enum import Enum
from typing import Union
from decimal import Decimal
from pydantic import BaseModel, ValidationError, validator
from pydantic.dataclasses import dataclass


class WhatWasIt(Enum):
    STR = 0
    INT = 1
    FLOAT = 2
    DEC = 3
    NONE = 4

class TheIdea(BaseModel):
    number: Decimal
    what_was_it: WhatWasIt = WhatWasIt.STR

    @classmethod
    def __get_validators__(cls):
        yield cls.validate

    @classmethod
    def validate(cls, v: Union["TheIdea", str, int, float, Decimal]) -> "TheIdea":
        if isinstance(v, TheIdea):
            return v  # Done validating
        if isinstance(v, str):
            return TheIdea(number=Decimal(v), what_was_it=WhatWasIt.STR)
        if isinstance(v, int):
            return TheIdea(number=Decimal(v), what_was_it=WhatWasIt.INT)
        if isinstance(v, float):
            return TheIdea(number=Decimal(v), what_was_it=WhatWasIt.FLOAT)
        if isinstance(v, Decimal):
            return TheIdea(number=v, what_was_it=WhatWasIt.DEC)

        raise ValueError("???")


@dataclass
class ParentDataClass:
    stuff: TheIdea


models = [
    ParentDataClass(stuff=TheIdea(number=Decimal("1.0"), what_was_it=WhatWasIt.NONE)),
    ParentDataClass(stuff="1e-3"),
    ParentDataClass(stuff=3),
    ParentDataClass(stuff=3.14159),
    ParentDataClass(stuff=Decimal(0)),
]

[print(m) for m in models]
```

Here `TheIdea` is the stand-in for `Prefixed`.  
Very cool!

**Comment by dan-fritchman** [comment]
Commit [d6e08a8](https://github.com/dan-fritchman/Hdl21/commit/d6e08a81fde7ec1972c0ba017f7ccf8f170824a0) has (I think) a complete implementation of this. 

In a paired change, the `Scalar` type gets more complicated than just `Union[Prefixed, str]`. Since we want most numeric-looking things to convert to `Prefixed` instead of `str`, it becomes a custom type, with similar pydantic-enabled validation machinery to `Prefixed`. It attempts to convert strings to numbers, and only if this fails leaves them as "expression literals". 

And that "expression literal", non-`Prefixed` variant of `Scalar` is also no longer the built-in string, but a new addition: `Literal`. As discussed in #65, this should probably become a new kind of Module attribute.

**Comment by dan-fritchman** [comment]
Merged in via #80
